numpyro/contrib/module.py

```diff
@@ -42,19 +42,19 @@ def flax_module(name, nn_module, *, input_shape=None):
     module_key = name + '$params'
     nn_params = numpyro.param(module_key)
     if nn_params is None:
-        if input_shape is None:
-            raise ValueError('Valid value for `input_shape` needed to initialize.')
+        if input_shape is not None:
+            args = (jnp.ones(input_shape),)
```

Comment:
nit: I guess `jnp.zeros(...)` is a better default input (at least for linear modules). What do you think? This might not be important though.

Reply:
I dont think the values are actually used. I'm not an expert in either flax/haiku, but it appears that its just dummy-data to bootstrap the initialized of nn.Module weights. The network weights are initialized according to their respective init plans.

Reply:
Thanks for explaining! That makes to me.